src/Kernel/ClassDescription.class.st

```diff
@@ -1369,6 +1369,19 @@ ClassDescription >> whichCategoryIncludesSelector: aSelector [
 		ifFalse: [^nil]
 ]
 
+{ #category : #'testing method dictionary' }
+ClassDescription >> whichSelectorsAccess: instVarName [
+	"Answer a set of selectors whose methods access the argument, 
+	instVarName, as a named instance variable."
+
+	| slot |
+	(self hasSlotNamed: instVarName) ifFalse: [ ^#() ].
+	slot := self slotNamed: instVarName.
+	^ self selectors select:  [:sel | | method |
+		method := self compiledMethodAt: sel.
+		(method readsSlot: slot) or: [method writesSlot: slot]]
```

Comment:
We already have polymorphic optimized methods in slot classes: #isReadIn: and #isWrittenIn: . #readsSlot is general method which is used by Slot class implementation. And readsField: is used by InstanceVariableSlot implementation.
So here method should use:
```Smalltalk
(slot isReadsIn: method) or: [slot isWrittenIn: method]
```
And it would be nice to add isReferencedIn: method for such cases.